Sources/SocksCore/Socket.swift

```diff
@@ -42,7 +42,10 @@ extension Socket {
         if config.reuseAddress {
             try setOption(descriptor: descriptor, level: SOL_SOCKET, name: SO_REUSEADDR, value: 1)
         }
-        
+
+        // prevents SIGPIPE from killing process
+        try setOption(descriptor: descriptor, level: SOL_SOCKET, name: SO_NOSIGPIPE, value: 1)
```

Comment:
I added it here because I was getting some sigpipes from both `send` and `accept`. I will test `0.5.0` and see if it still happens.